linode/manager
# upcoming: [M3-7714] - Use API filtering on `user_type` to populate the two Users & Grants tables

## Description 📝
When managing a child account with numerous users, the current approach to filtering users in a single query can lead to data discrepancies across table pages.

Take for example a child account with over 50 users, where the proxy user appears last. The proxy user won't be visible in its "Business Partner Settings" table until the third pagination page. 

To resolve this, we're using separate queries for each table, incorporating an API filter by `user_type`, ensuring the API supports this filter. This approach mirrors what we're doing for the Images landing page.

## Changes  🔄
- Added another `useAccountUsers` query with x-filter `{ user_type: 'proxy' }` for our proxy user (we will only ever have one).
- We no longer need `renderTableContent()` with this refactor since we have a component that handles this.
- Updated MSW to include filtering and ordering

## Target release date 🗓️
Tuesday, February 20th, otherwise Monday, March 4th.

## Preview 📷
| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/125309814/b2fcdc89-4489-4868-b39d-0cd84e7b34a3" /> | <video src="https://github.com/linode/manager/assets/125309814/0c5cdb1a-d9b4-46ee-99c2-f2422419dbbd" /> |

## How to test 🧪

### Prerequisites
(How to setup test environment)
- Run branch locally
- Turn on MSW and Parent/Child feature flag using dev tools
- Change [this line](https://github.com/linode/manager/blob/develop/packages/manager/src/mocks/serverHandlers.ts#L556) to `proxy`

### Reproduction steps
(How to reproduce the issue, if applicable)
- Go to http://localhost:3000/account/users

### Verification steps
(How to verify changes)
- Open dev tools and observe requests, data, and header x-filters

## As an Author I have considered 🤔

*Check all that apply*
